src/frontend/src/lib/canisters/signer.canister.ts

```diff
@@ -0,0 +1,98 @@
+import type {
+	BitcoinNetwork,
+	_SERVICE as SignerService,
+	SignRequest
+} from '$declarations/signer/signer.did';
+import { idlFactory as idlCertifiedFactorySigner } from '$declarations/signer/signer.factory.certified.did';
+import { idlFactory as idlFactorySigner } from '$declarations/signer/signer.factory.did';
+import { getAgent } from '$lib/actors/agents.ic';
+import type { BtcAddress, EthAddress } from '$lib/types/address';
+import type { Identity } from '@dfinity/agent';
+import { Principal } from '@dfinity/principal';
+import { Canister, createServices, type CanisterOptions } from '@dfinity/utils';
+
+export interface SignerCanisterOptions<T> extends Omit<CanisterOptions<T>, 'canisterId' | 'agent'> {
+	canisterId: Principal;
+	identity: Identity;
+}
+
+type SignerCanisterFunctionParams<T = unknown> = T & {
+	certified?: boolean;
+};
+
+export class SignerCanister extends Canister<SignerService> {
+	static async create({ identity, ...options }: SignerCanisterOptions<SignerService>) {
+		const agent = await getAgent({ identity });
+
+		const { service, certifiedService, canisterId } = createServices<SignerService>({
+			options: {
+				...options,
+				agent
+			},
+			idlFactory: idlFactorySigner,
+			certifiedIdlFactory: idlCertifiedFactorySigner
+		});
+
+		return new SignerCanister(canisterId, service, certifiedService);
+	}
+
+	getBtcAddress = ({
```

Comment:
Actually none of those methods, `getBtcAddress` , `getEthAddress` etc. are available as query. They are all always `certified = true`. Should we simplify their interfaces accordingly?

Reply:
Yes, sounds good! Adjusted.